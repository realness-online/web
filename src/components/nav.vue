<template>
  <nav id="main_nav" class='guest'>
    <router-link v-if="show" to="/profile" class="black">Name</router-link>
    <router-link v-if="show" to="/relationships" class="green">Relations</router-link>
    <router-link v-if="show" to="/groups" class="green">Groups</router-link>
    <router-link v-if="show" to="/events" class="blue">Events</router-link>
    <router-link v-if="show" to="/feed" class="blue">Feed</router-link>
    <wat-textarea class="red" v-on:toggle-keyboard="show = !show" tabindex="1"></wat-textarea>
  </nav>
</template>

<script>
  import wat_textarea from '@/components/wat-textarea'
  export default {
    components: {
      'wat-textarea': wat_textarea
    },
    data() {
      return {
        show: true
      }
    }
  }
</script>

<style lang="stylus">
  @require '../application'
  nav#main_nav
    min-height: 83vh
    display: flex
    flex-direction:row
    flex-wrap:wrap
    align-content: space-evenly
    justify-content:space-evenly
    align-items: flex-start
    & > *
      -webkit-tap-highlight-color: transparent
      animation-name: shimer
      animation-delay: 0.10s
      animation-duration: 0.25s
      transition-duration: 0
      transition-timing-function:ease-out
      text-align: left
      width: 44vw
      height: 24vh
      border-width: 1vmax
      border-style: solid
      border-radius: 3vw
      padding:base-line
    & > a
      &:nth-child(even)
        text-align: right
        padding-right:base-line
      &:active
        border-width: 1.33vh
        width: 42vw
        height: 22vh
        margin: 1vh 1vw
        color:transparent
      &:focus
        color:yellow
        transition-duration: 0.6s
        opacity:1
        transition: all
        margin:0.5vh 0.5vw
        width: 43vw
        height: 23vh
        outline: none

    &.guest
      & > [href='/profile']
      & > [href='/feed']
      & > [href='/relationships']
      & > [href='/events']
      & > [href='/groups']
        visibility: hidden

    &.guest-posts
      & > [href='/relationships']
      & > [href='/events']
      & > [href='/groups']
        visibility: hidden

    &.person
      & > [href='/events']
      & > [href='/groups']
        visibility: hidden

    &.has-ten-friends
      & > [href='/groups']
        visibility: hidden

     &.has-25-friends
       //
</style>
