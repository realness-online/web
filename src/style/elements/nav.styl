nav.profile-list
  display: grid
  grid-gap: base-line
  grid-template-columns: repeat(auto-fill, minmax((base-line * 12i), 1fr))
  @media (min-width: pad-begins)
    padding: 0 base-line
    & > figure
      standard-border: blue
      & > svg
        border-right-width: round((base-line / 10), 2)
        border-radius: 0
