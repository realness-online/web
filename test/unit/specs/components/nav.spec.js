import Vue from 'vue'
import {shallow} from 'vue-test-utils'
import nav from '@/components/nav'

describe('posts-list.vue', () => {

  it('should render the application navigation', () => {
    let wrapper = shallow(nav)
    expect(wrapper.element).toMatchSnapshot()
  })

})
