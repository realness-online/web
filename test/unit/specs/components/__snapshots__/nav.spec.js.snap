// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`posts-list.vue should render the application navigation 1`] = `
<nav
  class="guest_posts"
  id="main_nav"
>
  <router-link
    class="black"
    to="/profile"
  >
    Name
  </router-link>
   
  <router-link
    class="green"
    to="/relationships"
  >
    Relations
  </router-link>
   
  <router-link
    class="green"
    to="/groups"
  >
    Groups
  </router-link>
   
  <router-link
    class="blue"
    to="/events"
  >
    Events
  </router-link>
   
  <router-link
    class="blue"
    to="/feed"
  >
    Feed
  </router-link>
   
  <!---->
</nav>
`;
